### ISSUE TITLE
Removal of all commander rows on the settings tab causes unrecoverable error

### ISSUE THREAD

**Issue by danielbarla** [EDDiscovery/EDDiscovery#308]
On the settings tab, deleting the last commander row (thus leaving the grid empty) causes EDDiscovery to throw an uncaught exception.

Following this, the application cannot be started due to an immediate error:

![capture](https://cloud.githubusercontent.com/assets/12608979/17294421/0bb464cc-57f7-11e6-9237-9adfcd60f624.PNG)

Attempting to repair via the installer doesn't seem to fix problem, and neither does uninstalling / reinstalling EDDiscovery.  

Apologies for not having additional screenshots, I'm not yet to be able to restart EDDiscovery to replicate the start of the issue.

**Comment by danielbarla** [comment]
Update - removal of application data folder in C:\Users\[user]\AppData\Local\EDDiscovery\ seems to make EDDiscovery act as if it had a fresh install.

To reproduce the issue, click delete while the only commander is selected:

![capture](https://cloud.githubusercontent.com/assets/12608979/17294627/483e46b4-57f8-11e6-8a0c-8c608b53ee66.PNG)

After confirming, an error results:

![capture](https://cloud.githubusercontent.com/assets/12608979/17294684/8589f66c-57f8-11e6-905e-182fb701b53d.PNG)

Stack trace:

> ************** Exception Text **************
> System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than > the size of the collection.
> Parameter name: index
>    at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument,  ExceptionResource resource)
>    at System.Collections.Generic.List`1.get_Item(Int32 index)
>    at EDDiscovery2.EDDConfig.get_CurrentCommander()
>    at EDDiscovery.TravelHistoryControl.LoadCommandersListBox()
>    at EDDiscovery2.Settings.btnDeleteCommander_Click(Object sender, EventArgs e)
>    at System.Windows.Forms.Control.OnClick(EventArgs e)
>    at System.Windows.Forms.Button.OnClick(EventArgs e)
>    at System.Windows.Forms.Button.WndProc(Message& m)
>    at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr  lparam)

After this point, EDDiscovery cannot be started again.

**Comment by klightspeed** [comment]
Looks like we need to either handle the case where there are no commanders, or always ensure there is always at least one commander entry.

**Comment by danielbarla** [comment]
From my side, preventing the removal sounds like a good idea, and should fix the cause of issue in the first place.  It's probably easier than fixing all the places where the code will assume to use the first commander record.

As to how I got confused in the first place: I'm a new user, and ended up getting messing up during initial config.  The recently visited systems grid was not always populating upon re-start of the app, leading me to try to add a commander record (instead of just editing the first one).  I then had two records in the commander grid (IDs 0 and 1).  I clicked remove, and ended up removing item 0, after which the netlogs never seemed to load correctly.  I thought perhaps the app had some affinity for a record with an ID of 0, so I thought perhaps I could remove them all, and add a new one again (hopefully with an ID of 0).

Very cool app though, thanks for everyone involved.
